lib/hooks/useApiCallStatus.tsx

```diff
@@ -21,10 +22,22 @@ const useApiCallStatus = ({
   scrollToError,
   pathToPush,
   query,
+  pendingStatusStateDelay = 0,
 }: UseApiCallStatusProps) => {
   const router = useRouter();
+  const [pendingStatus, setPendingStatus] = useState<boolean>(false);
+  let timer: NodeJS.Timeout;
 
   useEffect(() => {
+    if (selector?.callStatus === ApiCallStatusCode.PENDING) {
+      timer = setTimeout(() => {
+        setPendingStatus(true);
+      }, pendingStatusStateDelay);
```

Comment:
Any chance of this launching multiple setTimeouts due to re-renders?

Igf so, perhaps, it would help to include the `pendingStatus` in the conditional to prevent it from happening multiple times?

```js
if (!pendingStatus && selector?.callStatus === ApiCallStatusCode.PENDING) { ... }
```

Reply:
Yeah that's a good point thanks.